### PR TITLE
Update to upstream v1.5.1

### PR DESCRIPTION
**Pull request by miyunari**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,127 +1,122 @@
 module github.com/sigstore/fulcio
 
-go 1.21
-
-// toolchain go1.21.3
+go 1.22.4
```
Comment (miyunari) [inline go.mod:3]:
```suggestion
go 1.21
```

File: go.mod [hunk 2]
```diff
@@ -1,127 +1,124 @@
 module github.com/sigstore/fulcio
 
-go 1.21
+go 1.22.2
 
-// toolchain go1.21.3
+toolchain go1.22.3
```
Comment (miyunari) [inline go.mod:5]:
````suggestion
go 1.21
````

**Comment by miyunari** [comment]
Can we upgrade Konflux to use go 1.22?
I tried to use an older go version, but:
```
go: chainguard.dev/sdk@v0.1.20 requires go >= 1.22.2 (running go 1.21.12; GOTOOLCHAIN=local)
```
:smile:

**Comment by lance** [comment]
I have opened a new PR bumping to 1.6.0 but we are still waiting for Go 1.22 in cachi2. I'll close this in favor of https://github.com/securesign/fulcio/pull/166

**Pull request closed without merging**